eq-author/src/components/modals/ConfirmationModal/index.js

```diff
@@ -73,16 +77,42 @@ const ConfirmationModal = ({
         <Button variant="secondary" onClick={handleCancel}>
           {cancelText}
         </Button>
-        <Button variant="primary" onClick={handleConfirm}>
+        <Button variant="primary" onClick={handleConfirm} autoFocus>
           {confirmText}
         </Button>
       </ButtonGroupStyled>
     </Modal>
   );
 };
 
+export const useConfirmationModal = ({ action, ...props }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleConfirm = useCallback(() => {
+    action();
+    setIsOpen(false);
+  }, [action]);
+
+  const handleCancel = useCallback(() => setIsOpen(false));
+
+  return {
+    trigger: useCallback(() => setIsOpen(true)),
```

Comment:
Same here

Reply:
Aye, I assumed no second param would imply [] but it seems not. Cheers! Will push in a sec.